core/notification/router_receiver_service.go

```diff
@@ -26,10 +28,26 @@ func (s *RouterReceiverService) PrepareMetaMessages(ctx context.Context, n Notif
 		return nil, nil, errors.ErrInvalid.WithMsgf("number of receiver selectors should be less than or equal threshold %d", s.deps.Cfg.MaxNumReceiverSelectors)
 	}
 
-	rcvs, err := s.deps.ReceiverService.List(ctx, receiver.Filter{
-		MultipleLabels: n.ReceiverSelectors,
-		Expanded:       true,
-	})
+	var rcvs []receiver.Receiver
+	var userConfigs map[uint64]map[string]interface{}
+
+	// Check if any selector contains a config
+	hasConfig := false
+	for _, selector := range n.ReceiverSelectors {
```

Comment:
what if there are multiple selectors have `config` key? this only handle 1 and first occurence

Reply:
For this, have 1 question
In case multiple selectors considering config with  multiple channels or passing channels in separate receivers . 
For example: 
Passing multiple channel
```
"receivers": [
    {
        "id": "1",
        "config": {
            "channel_name": "de-test", 
            "channel_name": "de-infra-integration"
        }
    }
]
```
Passing different channel in separate receivers.

```
"receivers": [
    {
        "id": "1",
        "config": {
            "channel_name": "de-test"
        }
    },
    {
        "id": "1",
        "config": {
            "channel_name": "de-infra-integration"
        }
    }
]
```
Second one is the correct payload. CMIIW

Reply:
The second one is the correct one.
Hm can we do cast the config to the specific receiver config directly instead of doing manual work like this?